dev/fattest.simplicity/src/componenttest/containers/ExternalTestServiceDockerClientStrategy.java

```diff
@@ -302,15 +308,65 @@ protected boolean isApplicable() {
         return true;
     }
 
-    public static boolean useRemoteDocker() {
-        if (USE_REMOTE_DOCKER) {
-            return true; // remote docker explicitly requested
-        }
-        if (Boolean.parseBoolean(System.getenv("GITHUB_ACTIONS"))) {
-            return false; // always use local docker for GH Actions
-        }
-        // Otherwise, use local docker for local runs, and remote docker for remote (RTC) runs
-        return !FATRunner.FAT_TEST_LOCALRUN;
-    }
+    /**
+     * Determines if we are going to attempt to run against a remote
+     * docker host, or a local docker host.
+     *
+     * Priority:
+     * 1. System Property: fat.test.use.remote.docker
+     * 2. System Property: fat.test.docker.host -> REMOTE
+     * 3. System: GITHUB_ACTIONS -> LOCAL
+     * 4. System: WINDOWS -> REMOTE
+     *
+     * default (!!! fat.test.localrun)
+     *
+     * @return true, we are running against a remote docker host, false otherwise.
+     */
+    private static boolean useRemoteDocker() {
+        boolean result;
+        String reason;
+
+        do {
+            //State 1: fat.test.use.remote.docker should always be honored first
+            if (System.getProperty("fat.test.use.remote.docker") != null) {
+                result = Boolean.getBoolean("fat.test.use.remote.docker");
+                reason = "fat.test.use.remote.docker set to " + result;
+                break;
+            }
+
+            //State 2: User provided a remote docker host, assume they want to use the remote host
+            if (USE_DOCKER_HOST != null) {
+                result = true;
+                reason = "fat.test.docker.host set to " + USE_DOCKER_HOST;
```

Comment:
Can "fat.test.docker.host" be set to the localhost?  Or is it pretty safe that a value in this field will only be the name of a remote host?

Reply:
It could be set to localhost and the following would happen:
1. This would return true
2. Then the ExternalTestServiceDockerClientStategy (this class) would get used first as it has priority 900
3. None of the remote docker hosts set in user.build.properties would match "localhost"
4. We would try the next docker strategy with a priority
5. Either find a docker host locally and run, or fail to find a local docker host. 

But this value is only set by a user or a build system so it's pretty safe that we won't be doing the above process all the time.